quora-service/src/main/java/com/upgrad/quora/service/business/QuestionBusinessService.java

```diff
@@ -0,0 +1,118 @@
+package com.upgrad.quora.service.business;
+
+import com.upgrad.quora.service.dao.QuestionDao;
+import com.upgrad.quora.service.dao.UserDao;
+import com.upgrad.quora.service.entity.AnswerEntity;
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.entity.UserAuthEntity;
+import com.upgrad.quora.service.entity.UserEntity;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.InvalidQuestionException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Service
+public class QuestionBusinessService {
+
+    @Autowired
+    private UserDao userDao;
+
+    @Autowired
+    private QuestionDao questionDao;
+
+    @Transactional(propagation = Propagation.REQUIRED, rollbackFor = InvalidQuestionException.class)
+    public QuestionEntity createQuestion(String authorization, QuestionEntity questionEntity) throws AuthorizationFailedException, InvalidQuestionException {
+        UserAuthEntity userAuthEntity = userDao.getUserAuthByToekn(authorization);
+        if (userAuthEntity == null) {
+            throw new AuthorizationFailedException("ATHR-001", "User has not signed in");
+        } else if (userAuthEntity.getLogoutAt() != null) {
+            throw new AuthorizationFailedException("ATHR-002", "User is signed out");
+        }
+
+        if (questionEntity == null) {
```

Comment:
questionEntity won't be null here , because we are sending the new question entity object from api method .
please check once .